dashpay/src/main/kotlin/org/dashj/platform/contracts/wallet/TxMetadata.kt

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2020-present, Dash Core Group
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package org.dashj.platform.contracts.wallet
```

Comment:
this is under a package called "contracts" and "wallet"  -- "wallet" is labeled as "dashwallet" in other places.

"dashwallet" could get renamed to "wallet" where the contract applies only to a wallet and the things that the wallet can contain.

A wallet contains transactions, so the transaction metadata is in this contract.  Another thing that could be included in this contract would be a document that defines the wallet creation time (encrypted) and possibly a checkpoint on the blockchain.  Knowing this time would speed up restoring a recovery phrase.  

Our other important contract is dashpay, which is related to higher level functionality where there are usernames, contact requests, and profiles and contact info (not yet implemented).